Add blog entry for v0.13.0 (#114)
shipwright-io/website: content/en/blog/posts/2024-03-27-build-release-v0.13.0.md

### Beta API Changes

We've implemented minor adjustments to the existing `v1beta1` API for the sake of consistency, incorporating feedback gathered from the `v0.12.0` release. These are:

- Renaming certain Go Types, such as `build.Spec.Source.GitSource` to `build.Spec.Source.Git`, aligning them with their respective JSON tags.
- Designating `build.Spec.Source.Type` and `buildRun.Spec.Source.Type` as mandatory, addressing usability concerns raised by users.
- We made `build.Spec.Source` optional. Enabling the definition of a Build without any source. This feature proves valuable
  particularly when executing a Build only with local source.
- Requiring `build.spec.source.git.url` upon the specification of `build.spec.source.git`.

Further details can be found in [PR 1504](https://github.com/shipwright-io/build/pull/1504) and [PR 1441](https://github.com/shipwright-io/build/pull/1441).

## Bugs

### Conversion Webhook

As the usage of the webhook increased over the last months, we've made enhancements to address certain gaps:

> - Previously, there was an issue with the proper conversion of BuildRuns from v1alpha1 to v1beta1 when a generated service account was utilized. This has been resolved.

Reviewer: ```suggestion
- Previously, there was an issue with the proper conversion of `BuildRun` instances from v1alpha1 to v1beta1 when a generated service account was utilized. This has been resolved.
```
Generally, put object names between backticks and add a label like "instances" or whatever accurately describes them. Make the label singular or plural, as needed. Don't make object names plural unless they are also plural in your code.

Author: great advice, highly appreciated!

- Additionally, a bug was identified where patching a completed v1beta1 BuildRun would inadvertently remove its status. This issue has been resolved.

### OCIArtifacts

> When implementing prevention measures against path traversal during the extraction of an OCI artifact, we were too strict. We only needed to prevent /../ because this means to go one directory up. We still must allow `..` because a directory or file can contain two subsequent dots in its name. You can now use files and directories with two subsequent dots in its name when using an OCI artifact as source.

Reviewer: ```suggestion
When implementing prevention measures against path traversal during the extraction of an OCI artifact, we were too strict. We only needed to prevent /../ because this means to go one directory up. We still must allow `..` because a directory or file can contain two subsequent dots in its name. You can now use files and directories with two subsequent dots in their names when using an OCI artifact as source.
```

> When implementing prevention measures against path traversal during the extraction of an OCI artifact, we were too strict. We only needed to prevent /../ because this means to go one directory up. We still must allow `..` because a directory or file can contain two subsequent dots in its name. You can now use files and directories with two subsequent dots in its name when using an OCI artifact as source.

Reviewer: ```suggestion
When implementing prevention measures against path traversal during the extraction of an OCI artifact, we were too strict. We only needed to prevent `/../` because this means going up one directory. We still must allow `..` because a directory or file can contain two subsequent dots in its name. You can now use files and directories with two subsequent dots in its name when using an OCI artifact as source.
```
